x-pack/filebeat/input/httpjson/pagination.go

```diff
@@ -65,6 +65,12 @@ func newPagination(config config, httpClient *httpClient, log *logp.Logger) *pag
 	return pagination
 }
 
+func newChainPagination(config chainConfig, httpClient *httpClient, log *logp.Logger) *pagination {
+	pagination := &pagination{httpClient: httpClient, log: log}
+
+	return pagination
+}
```

Comment:
addressed

Reply:
I don't think we need this, it's only called from one site and doesn't do any validation. I would replace the call with `&pagination{httpClient: httpClient, log: log}`.

Reply:
@efd6 this was done with the intention that in future when we add support for pagination within chain blocks , we could expand upon the existing method , would like to hear your opinion on that approach.

Reply:
Until it's needed for doing that, it doesn't really need to be here. There is only one site that it's called at, so it will be easy to add when the time comes.